### DISCUSSION TITLE
OpenShift 3.11 remove service-account-private-key-file

### DISCUSSION THREAD

**Discussion by adshafqat** [aquasecurity/kube-bench#844] [Questions and Help] [answered]
We are running kube-bench v3 tests in an OpenShift 3.11 environment and I am not sure how to find solution of this test failure

3.4 Edit the Openshift master config file /etc/origin/master/master-config.yaml and remove service-account-private-key-file

As per my understanding the test is failing because the privateKeyFile is set to the default value of serviceaccounts.private.key.

The serviceaccounts.private.key file is used by cluster to enable service account token generation and authentication. The file is currently stored in /etc/origin/master folder and the value is configured under in master-config.yaml file.

OpenShift 3.11 master-config.yaml file
serviceAccountConfig:
  masterCA: ca-bundle.crt
  privateKeyFile: serviceaccounts.private.key
  publicKeyFiles:
  - serviceaccounts.public.key

I am not sure what is the suggestion here? We can't remove this file as its required for Service account authentication. What are the possible solutions to pass this test?

**Comment by yoavrotems** [comment, marked as answer]
The remediation says: 
`Edit the Openshift master config file /etc/origin/master/master-config.yaml and remove service-account-private-key-file` 
If I understood correctly you can't delete it because you are using it.
Then just leave it, this is a recommendation, 
![image](https://user-images.githubusercontent.com/12915002/121900178-d1157980-cd2d-11eb-887d-5744d15781fa.png)

Maybe we should change it to be not scored test, then if it pass it PASS and if it fails it will be WARN which indicates "Hey something is needs your attention check it out".
